include/nuttx/net/netdev_lowerhalf.h

```diff
@@ -61,10 +62,8 @@
 
 #define NETPKT_BUFLEN   CONFIG_IOB_BUFSIZE
 
-#if defined(CONFIG_OPENAMP) || \
-    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L && \
-    !defined(__STDC_NO_ATOMICS__))
-#define HAVE_ATOMIC
+#if defined(CONFIG_OPENAMP) || defined(CONFIG_HAVE_ATOMICS)
```

Comment:
remove, let's replace HAVE_ATOMICS with CONFIG_HAVE_ATOMICS

Reply:
We can't since atomics can be provided by OpenAMP and not by the toolchain

Reply:
I think we can drop CONFIG_OPENAMP since the wrapper in CONFIG_OPENAMP also depends on c/c++ atomic. It's also not good to define the common macro(HAVE_ATOMIC) in the header file.

Reply:
Let me check `metal/atomic.h` and related code in OpenAMP

Reply:
Here it is a bit tricky as:
1. `metal/config.h` contains `#undef HAVE_STDATOMIC_H` due to `s/cmakedefine/undef/g"` from `liberal.defs`
2. `metal/atomic.h` will fallback to `#else` if `__cplusplus` is not defined so it will not pass `#elif defined(HAVE_STDATOMIC_H) && !defined(__CC_ARM) && !defined(__STDC_NO_ATOMICS__)` due to 1 above.
3. The `__GNUC__` is most probably defined, so it will go to `#elif defined(__GNUC__)` and `# include <metal/compiler/gcc/atomic.h>`. The `gcc/atomic.h` has implementation of atomic operation based on GNU extensions from https://gcc.gnu.org/onlinedocs/gcc-4.1.0/gcc/Atomic-Builtins.html that most probably are available even before GCC adopted C11 (but here I'm not sure).

Reply:
The other option is to add `nuttx/atomic.h` and base it on stdatomic.h if available or re-implement atomic types based on GNU extensions or spinlocks if stdatomic.h is not supported.

Reply:
you may steal stuff from there (still on going work) https://github.com/mu578/mu0/blob/master/mu0/mu0_definition/mu0_atomic.h gcc deprecated the sync* in favor of atomic* anyway same.

Reply:
Just checked that C11 is adopted by GCC starting from version 4.6 while sync builtin's are there at least from 4.1. So the question here is: should we target GCC versions between 4.1 and 4.6, or use either stdatomics or spinlock based protection?

Reply:
@pkarashchenko I had a hard to retrieve ; that discussion ; anyway I did a rally about such today: https://github.com/mu578/mu0/blob/master/mu0/mu0_definition/mu0_atomic.h
c11 primitives are very sparse ; so here a complete model of my digging. I think full support is 4.5 before that the sync* remains true.

Reply:
@pkarashchenko I did push this morning: c11 atomics only support fetch_and_op and not op_and_fetch, so I added them ; I don't know yet if I am getting that right, I need to dump assembly to see the transform ; then compare outputs which is a tedious and time consuming process if handled correctly.